diff --git a/src/asset-relocator.js b/src/asset-relocator.js
index badf9e086b653d5b0e46f5a758a0926268ff2300..c849534186bf2990aa1053d839eaab796c643730 100644
--- a/src/asset-relocator.js
+++ b/src/asset-relocator.js
@@ -342,7 +342,7 @@ function injectPathHook (compilation, outputAssetBase) {
     }
 
     generate() {
-      const requireBase = `${esm ? "new URL('.', import.meta.url).pathname.slice(import.meta.url.match(/^file:\\/\\/\\/\\w:/) ? 1 : 0, -1)" : '__dirname'} + ${JSON.stringify(this.relBase + '/' + assetBase(outputAssetBase))}`;
+      const requireBase = `${esm ? "decodeURIComponent(new URL('.', import.meta.url).pathname).slice(import.meta.url.match(/^file:\\/\\/\\/\\w:/) ? 1 : 0, -1)" : '__dirname'} + ${JSON.stringify(this.relBase + '/' + assetBase(outputAssetBase))}`;
 
       return `if (typeof __webpack_require__ !== 'undefined') __webpack_require__.ab = ${requireBase};`
     }
